src/screens/Survey/redux/containers/Container_Question.js

```diff
@@ -0,0 +1,28 @@
+import {connect} from 'react-redux';
+import Question from '../../Question';
+import {writeResponsesToDatabase,loadQuestionResponse,updateTotalQuestions,addKeyToSet,increaseIndex,decreaseIndex,setMaxIndex, SURVEY_NAMESPACE} from '../actions/Actions_question';
+import {getSurveyQuestions} from '../../../../match/redux'
+
+function mapStateToProps(state){
+  return {
+    questionResponses : state[SURVEY_NAMESPACE].questionResponses,
+    index : state[SURVEY_NAMESPACE].index,
+    totalNumQuestions: state[SURVEY_NAMESPACE].totalNumQuestions,
```

Comment:
Could this be calculated rather than stored just by looking at the length of the questions array?

Reply:
Yes, I think looking at the length of the array is a much better way to do it. I'm a huge fan of storing a minimum amount of state and then having relatively more complex selectors to act on the state and extract more information from it

Reply:
Cool, I think I got it that way now!